cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -28,6 +28,7 @@
 
 #include "arrow/dataset/discovery.h"
 #include "arrow/dataset/file_base.h"
+#include "arrow/dataset/parquet_encryption_config.h"
```

Comment:
It would be better to use forward declaration here and only include this header file in the file_parquet.cc

Reply:
I think so, it is generally the convention to use forward declaration whenever possible.